tools/swig/CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+# CMake configuration for generating script language interfaces with SWIG
+
+# set minimum CMake version required and switch to new policies for SWIG
+cmake_minimum_required(VERSION 3.14)
+if(POLICY CMP0078)
+  cmake_policy(SET CMP0078 NEW)
+endif()
+if(POLICY CMP0086)
+  cmake_policy(SET CMP0086 NEW)
+endif()
+
+# some of the find_package() scripts trigger developer warnings
+# even though they are bundled with CMake; Suppress them
+set(CMAKE_SUPPRESS_DEVELOPER_WARNINGS TRUE)
+
+if(NOT BUILD_SHARED_LIBS)
+  error(FATAL_ERROR "Option BUILD_SHARED_LIBS must be enabled to use SWIG wrappers")
+endif()
+
+find_package(SWIG REQUIRED)
+include(${SWIG_USE_FILE})
+
+option(BUILD_SWIG_JAVA "Build Java JNI wrapper with SWIG" OFF)
+option(BUILD_SWIG_LUA "Build Lua wrapper with SWIG" OFF)
+option(BUILD_SWIG_PERL5 "Build Perl5 wrapper with SWIG" OFF)
+option(BUILD_SWIG_PYTHON "Build Python wrapper with SWIG" OFF)
+option(BUILD_SWIG_RUBY "Build Ruby wrapper with SWIG" OFF)
+option(BUILD_SWIG_TCL "Build Tcl wrapper with SWIG" OFF)
+
+add_custom_target(clean-all
+  COMMAND ${CMAKE_COMMAND} --build . --target clean
+  COMMAND ${CMAKE_COMMAND} -E remove example.* *.class *.java *.pm log.lammps
```

Comment:
That seems un-cmakeish, use something like:
```
set_property(DIRECTORY APPEND PROPERTY ADDITIONAL_MAKE_CLEAN_FILES example.* *.class *.java *.pm log.lammps)
```
instead to append to the normal `clean` target.

Reply:
i was looking for something similar, but Google and the labyrinthine CMake failed to enlighten me.
unfortunately that directive does not seem to like wildcards. so I had to add all files explicitly.